keras/preprocessing/sequence.py

```diff
@@ -216,3 +218,95 @@ def _remove_long_seq(maxlen, seq, label):
             new_seq.append(x)
             new_label.append(y)
     return new_seq, new_label
+
+
+class TimeseriesGenerator(Sequence):
+    """Utility class for generating batches of temporal data.
+
+    This class takes in a sequence of data-points gathered at 
+    equal intervals, along with time series parameters such as 
+    stride, delay etc., to produce batches for training/validation.
+
+    # Arguments
+        data: indexable generator (such as list or Numpy array)
+            containing consecutive data points.
+        targets: targets corresponding to instances in `data`.
+            Should have same length as `data`.
+        lookback: length of history to consider for prediction.
+            `data[i-lookback : i+1]` is used to prepare the input.
+        delay: how further in the future we are looking to predict.
```

Comment:
This is actually somewhat complicated. Arguably the user should have baked that into `targets`, e.g. if I'm doing timeseries forecasting for `t + 10`, I would use `data=data[:-10, :, :], targets=data[10:, :, :]`. Thoughts?

Reply:
It would make the API a little less complex, yes. I think `delay` is useful as a convenience if the user wants to utilize one large array to train multiple models, with different delays. I don't see an issue with keeping it there, but open to taking it off if you see some strong reason to do otherwise.

Reply:
The more I think about it, the more it makes sense to let the user mutate `targets` instead of us using `delay`. They will usually have to perform some transformations themselves (in which case a simple temporal shift is no big deal), or the input & target reference indices will be the same (in which case delay=0).

Reply:
Ok, let's just use `targets` then. As long as it is clearly documented, it's a user-friendly API.